### PR TITLE
Revert "Guard use of struct tms with #ifdef __TMS"

### LINKED ISSUE #11903: 90% drop in aes performance since commit db71d315479762eefbf2bcda8be3b44b1867133f
Hi,

I've observed a 90% drop in aes performance since commit db71d315479762eefbf2bcda8be3b44b1867133f. I tested the previous commit c6601bd2d728d4c61711a016c6267fb45910e7cd, and the latest master commit e1c6f76281473b8fe66954187e793108a0e8568c with 5 benchmarks and two systems, ARM Ampere eMag 32 core and x86 i7-8550U, and found that all benchmarks dropped 75% to 90%+ on commit db71d315479762eefbf2bcda8be3b44b1867133f. The performance is still bad on the latest commit e1c6f76281473b8fe66954187e793108a0e8568c and I don't think an issue has been submitted for this yet.

System: ARM Ampere eMag 32 core  

c6601bd2d728d4c61711a016c6267fb45910e7cd  - Commit before performance drop-off
```
openssl speed -multi 1 -evp aes256-wrap
Forked child 0
Got: +H:16:64:256:1024:8192:16384 from 0
Got: +F:22:id-aes256-wrap:9132949.33:9824149.33:9924096.00:9917440.00:9912320.00:9912320.00 from 0
version: 3.0.0-alpha3-dev
built on: built on: Thu May 21 21:51:07 2020 UTC
options:bn(64,64) rc4(char) des(int) aes(partial) idea(int) blowfish(ptr)
compiler: gcc -fPIC -pthread -Wa,--noexecstack -Wall -O3 -DOPENSSL_USE_NODELETE -DOPENSSL_PIC -DOPENSSL_BUILDING_OPENSSL -DNDEBUG
CPUINFO: OPENSSL_armcap=0x3f
evp               9132.95k     9824.15k     9924.10k     9917.44k     9912.32k     9912.32k

openssl speed -multi 1 -evp id-aes256-wrap-pad
Forked child 0
Got: +H:16:64:256:1024:8192:16384 from 0
Got: +F:22:id-aes256-wrap-pad:9566874.67:9854677.33:9912149.33:9910954.67:9906858.67:9906858.67 from 0
version: 3.0.0-alpha3-dev
built on: built on: Thu May 21 21:51:07 2020 UTC
options:bn(64,64) rc4(char) des(int) aes(partial) idea(int) blowfish(ptr)
compiler: gcc -fPIC -pthread -Wa,--noexecstack -Wall -O3 -DOPENSSL_USE_NODELETE -DOPENSSL_PIC -DOPENSSL_BUILDING_OPENSSL -DNDEBUG
CPUINFO: OPENSSL_armcap=0x3f
evp               9566.87k     9854.68k     9912.15k     9910.95k     9906.86k     9906.86k

openssl speed -multi 1 -evp id-aes128-wrap-pad
Forked child 0
Got: +H:16:64:256:1024:8192:16384 from 0
Got: +F:22:id-aes128-wrap-pad:12592592.00:13065920.00:13169493.33:13165568.00:13156352.00:13156352.00 from 0
version: 3.0.0-alpha3-dev
built on: built on: Thu May 21 21:51:07 2020 UTC
options:bn(64,64) rc4(char) des(int) aes(partial) idea(int) blowfish(ptr)
compiler: gcc -fPIC -pthread -Wa,--noexecstack -Wall -O3 -DOPENSSL_USE_NODELETE -DOPENSSL_PIC -DOPENSSL_BUILDING_OPENSSL -DNDEBUG
CPUINFO: OPENSSL_armcap=0x3f
evp              12592.59k    13065.92k    13169.49k    13165.57k    13156.35k    13156.35k

openssl speed -multi 1 -evp des-ede3-cfb1
Forked child 0
Got: +H:16:64:256:1024:8192:16384 from 0
Got: +F:22:des-ede3-cfb1:293029.33:293333.33:293376.00:293546.67:292958.94:293932.23 from 0
version: 3.0.0-alpha3-dev
built on: built on: Thu May 21 21:51:07 2020 UTC
options:bn(64,64) rc4(char) des(int) aes(partial) idea(int) blowfish(ptr)
compiler: gcc -fPIC -pthread -Wa,--noexecstack -Wall -O3 -DOPENSSL_USE_NODELETE -DOPENSSL_PIC -DOPENSSL_BUILDING_OPENSSL -DNDEBUG
CPUINFO: OPENSSL_armcap=0x3f
evp                293.03k      293.33k      293.38k      293.55k      292.96k      293.93k

openssl speed -multi 1 -evp des-ede-ecb
Forked child 0
Got: +H:16:64:256:1024:8192:16384 from 0
Got: +F:22:des-ede:20772650.67:21217600.00:21376256.00:21414570.67:21427541.33:21430272.00 from 0
version: 3.0.0-alpha3-dev
built on: built on: Thu May 21 21:51:07 2020 UTC
options:bn(64,64) rc4(char) des(int) aes(partial) idea(int) blowfish(ptr)
compiler: gcc -fPIC -pthread -Wa,--noexecstack -Wall -O3 -DOPENSSL_USE_NODELETE -DOPENSSL_PIC -DOPENSSL_BUILDING_OPENSSL -DNDEBUG
CPUINFO: OPENSSL_armcap=0x3f
evp              20772.65k    21217.60k    21376.26k    21414.57k    21427.54k    21430.27k
```

db71d315479762eefbf2bcda8be3b44b1867133f  - Performance dropped off 75%-90%+
```
openssl speed -multi 1 -evp aes256-wrap
Forked child 0
Got: +H:16:64:256:1024:8192:16384 from 0
Got: +F:22:id-aes256-wrap:-124.63:-49447.45:49377.43:14853.37:628.65:456.77 from 0
version: 3.0.0-alpha3-dev
built on: built on: Thu May 21 22:07:18 2020 UTC
options:bn(64,64) rc4(char) des(int) aes(partial) idea(int) blowfish(ptr)
compiler: gcc -fPIC -pthread -Wa,--noexecstack -Wall -O3 -DOPENSSL_USE_NODELETE -DOPENSSL_PIC -DOPENSSL_BUILDING_OPENSSL -DNDEBUG
CPUINFO: OPENSSL_armcap=0x3f
evp               -124.63    -49447.45        49.38k       14.85k      628.65       456.77

openssl speed -multi 1 -evp id-aes256-wrap-pad
Forked child 0
Got: +H:16:64:256:1024:8192:16384 from 0
Got: +F:22:id-aes256-wrap-pad:-137.72:-99512.24:26959.14:5402.66:1193.45:906.03 from 0
version: 3.0.0-alpha3-dev
built on: built on: Thu May 21 22:07:18 2020 UTC
options:bn(64,64) rc4(char) des(int) aes(partial) idea(int) blowfish(ptr)
compiler: gcc -fPIC -pthread -Wa,--noexecstack -Wall -O3 -DOPENSSL_USE_NODELETE -DOPENSSL_PIC -DOPENSSL_BUILDING_OPENSSL -DNDEBUG
CPUINFO: OPENSSL_armcap=0x3f
evp               -137.72    -99512.24        26.96k     5402.66      1193.45       906.03

openssl speed -multi 1 -evp id-aes128-wrap-pad
Forked child 0
Got: +H:16:64:256:1024:8192:16384 from 0
Got: +F:22:id-aes128-wrap-pad:-183.11:-98689.45:-407232.66:12728.22:1844.09:969.68 from 0
version: 3.0.0-alpha3-dev
built on: built on: Thu May 21 22:07:18 2020 UTC
options:bn(64,64) rc4(char) des(int) aes(partial) idea(int) blowfish(ptr)
compiler: gcc -fPIC -pthread -Wa,--noexecstack -Wall -O3 -DOPENSSL_USE_NODELETE -DOPENSSL_PIC -DOPENSSL_BUILDING_OPENSSL -DNDEBUG
CPUINFO: OPENSSL_armcap=0x3f
evp               -183.11    -98689.45   -407232.66        12.73k     1844.09       969.68

openssl speed -multi 1 -evp des-ede3-cfb1
Forked child 0
Got: +H:16:64:256:1024:8192:16384 from 0
Got: +F:22:des-ede3-cfb1:-4.42:166.98:28.92:3.30:-0.01:0.18 from 0
version: 3.0.0-alpha3-dev
built on: built on: Thu May 21 22:07:18 2020 UTC
options:bn(64,64) rc4(char) des(int) aes(partial) idea(int) blowfish(ptr)
compiler: gcc -fPIC -pthread -Wa,--noexecstack -Wall -O3 -DOPENSSL_USE_NODELETE -DOPENSSL_PIC -DOPENSSL_BUILDING_OPENSSL -DNDEBUG
CPUINFO: OPENSSL_armcap=0x3f
evp                 -4.42       166.98        28.92         3.30        -0.01         0.18

openssl speed -multi 1 -evp des-ede-ecb
Forked child 0
Got: +H:16:64:256:1024:8192:16384 from 0
Got: +F:22:des-ede:-300.09:-79851.36:-661065.90:18878.85:2934.88:1215.26 from 0
version: 3.0.0-alpha3-dev
built on: built on: Thu May 21 22:07:18 2020 UTC
options:bn(64,64) rc4(char) des(int) aes(partial) idea(int) blowfish(ptr)
compiler: gcc -fPIC -pthread -Wa,--noexecstack -Wall -O3 -DOPENSSL_USE_NODELETE -DOPENSSL_PIC -DOPENSSL_BUILDING_OPENSSL -DNDEBUG
CPUINFO: OPENSSL_armcap=0x3f
evp               -300.09    -79851.36   -661065.90        18.88k     2934.88      1215.26
```

e1c6f76281473b8fe66954187e793108a0e8568c  - Latest master commit. Performance is still bad.
```
openssl speed -multi 1 -evp aes256-wrap
Forked child 0
Got: +H:16:64:256:1024:8192:16384 from 0
Got: +F:22:id-aes256-wrap:-120.07:-299739.05:16504.26:3763.91:453.29:368.02 from 0
version: 3.0.0-alpha3-dev
built on: built on: Thu May 21 22:27:15 2020 UTC
options:bn(64,64) rc4(char) des(int) aes(partial) idea(int) blowfish(ptr)
compiler: gcc -fPIC -pthread -Wa,--noexecstack -Wall -O3 -DOPENSSL_USE_NODELETE -DOPENSSL_PIC -DOPENSSL_BUILDING_OPENSSL -DNDEBUG
CPUINFO: OPENSSL_armcap=0x3f
evp               -120.07   -299739.05        16.50k     3763.91       453.29       368.02

openssl speed -multi 1 -evp id-aes256-wrap-pad
Forked child 0
Got: +H:16:64:256:1024:8192:16384 from 0
Got: +F:22:id-aes256-wrap-pad:-141.51:-59468.36:98140.94:5943.01:1508.43:1243.38 from 0
version: 3.0.0-alpha3-dev
built on: built on: Thu May 21 22:27:15 2020 UTC
options:bn(64,64) rc4(char) des(int) aes(partial) idea(int) blowfish(ptr)
compiler: gcc -fPIC -pthread -Wa,--noexecstack -Wall -O3 -DOPENSSL_USE_NODELETE -DOPENSSL_PIC -DOPENSSL_BUILDING_OPENSSL -DNDEBUG
CPUINFO: OPENSSL_armcap=0x3f
evp               -141.51    -59468.36        98.14k     5943.01      1508.43      1243.38

openssl speed -multi 1 -evp id-aes128-wrap-pad
Forked child 0
Got: +H:16:64:256:1024:8192:16384 from 0
Got: +F:22:id-aes128-wrap-pad:-186.48:-78818.77:65497.37:17146.55:1252.87:926.44 from 0
version: 3.0.0-alpha3-dev
built on: built on: Thu May 21 22:27:15 2020 UTC
options:bn(64,64) rc4(char) des(int) aes(partial) idea(int) blowfish(ptr)
compiler: gcc -fPIC -pthread -Wa,--noexecstack -Wall -O3 -DOPENSSL_USE_NODELETE -DOPENSSL_PIC -DOPENSSL_BUILDING_OPENSSL -DNDEBUG
CPUINFO: OPENSSL_armcap=0x3f
evp               -186.48    -78818.77        65.50k       17.15k     1252.87       926.44

openssl speed -multi 1 -evp des-ede3-cfb1
Forked child 0
Got: +H:16:64:256:1024:8192:16384 from 0
Got: +F:22:des-ede3-cfb1:-4.35:49.16:93.60:4.97:0.56:0.56 from 0
version: 3.0.0-alpha3-dev
built on: built on: Thu May 21 22:27:15 2020 UTC
options:bn(64,64) rc4(char) des(int) aes(partial) idea(int) blowfish(ptr)
compiler: gcc -fPIC -pthread -Wa,--noexecstack -Wall -O3 -DOPENSSL_USE_NODELETE -DOPENSSL_PIC -DOPENSSL_BUILDING_OPENSSL -DNDEBUG
CPUINFO: OPENSSL_armcap=0x3f
evp                 -4.35        49.16        93.60         4.97         0.56         0.56

openssl speed -multi 1 -evp des-ede-ecb
Forked child 0
Got: +H:16:64:256:1024:8192:16384 from 0
Got: +F:22:des-ede:-306.20:-79872.80:159130.36:-216305.35:15667.22:1728.11 from 0
version: 3.0.0-alpha3-dev
built on: built on: Thu May 21 22:27:15 2020 UTC
options:bn(64,64) rc4(char) des(int) aes(partial) idea(int) blowfish(ptr)
compiler: gcc -fPIC -pthread -Wa,--noexecstack -Wall -O3 -DOPENSSL_USE_NODELETE -DOPENSSL_PIC -DOPENSSL_BUILDING_OPENSSL -DNDEBUG
CPUINFO: OPENSSL_armcap=0x3f
evp               -306.20    -79872.80       159.13k  -216305.35        15.67k     1728.11
```

System: x86 i7-8550U  

c6601bd2d728d4c61711a016c6267fb45910e7cd  - Commit before performance drop-off
```
openssl speed -multi 1 -evp aes256-wrap
Forked child 0
Got: +H:16:64:256:1024:8192:16384 from 0
Got: +F:22:id-aes256-wrap:9137258.67:9128725.33:9134592.00:9243648.00:8719018.67:9049429.33 from 0
version: 3.0.0-alpha3-dev
built on: built on: Thu May 21 21:54:40 2020 UTC
options:bn(64,64) rc4(16x,int) des(int) aes(partial) idea(int) blowfish(ptr)
compiler: gcc -fPIC -pthread -m64 -Wa,--noexecstack -Wall -O3 -DOPENSSL_USE_NODELETE -DL_ENDIAN -DOPENSSL_PIC -DOPENSSL_BUILDI
NG_OPENSSL -DNDEBUG
CPUINFO: OPENSSL_ia32cap=0xfedaf387ffebffff:0x9c67ab
evp 9137.26k 9128.73k 9134.59k 9243.65k 8719.02k 9049.43k


openssl speed -multi 1 -evp id-aes256-wrap-pad
Forked child 0
Got: +H:16:64:256:1024:8192:16384 from 0
Got: +F:22:id-aes256-wrap-pad:8662757.33:9154560.00:9050538.67:9133331.56:9232384.00:9262421.33 from 0
version: 3.0.0-alpha3-dev
built on: built on: Thu May 21 21:54:40 2020 UTC
options:bn(64,64) rc4(16x,int) des(int) aes(partial) idea(int) blowfish(ptr)
compiler: gcc -fPIC -pthread -m64 -Wa,--noexecstack -Wall -O3 -DOPENSSL_USE_NODELETE -DL_ENDIAN -DOPENSSL_PIC -DOPENSSL_BUILDI
NG_OPENSSL -DNDEBUG
CPUINFO: OPENSSL_ia32cap=0xfedaf387ffebffff:0x9c67ab
evp 8662.76k 9154.56k 9050.54k 9133.33k 9232.38k 9262.42k


openssl speed -multi 1 -evp id-aes128-wrap-pad
Forked child 0
Got: +H:16:64:256:1024:8192:16384 from 0
Got: +F:22:id-aes128-wrap-pad:12268282.67:12572672.00:12626773.33:12727978.67:12926976.00:12899669.33 from 0
version: 3.0.0-alpha3-dev
built on: built on: Thu May 21 21:54:40 2020 UTC
options:bn(64,64) rc4(16x,int) des(int) aes(partial) idea(int) blowfish(ptr)
compiler: gcc -fPIC -pthread -m64 -Wa,--noexecstack -Wall -O3 -DOPENSSL_USE_NODELETE -DL_ENDIAN -DOPENSSL_PIC -DOPENSSL_BUILDI
NG_OPENSSL -DNDEBUG
CPUINFO: OPENSSL_ia32cap=0xfedaf387ffebffff:0x9c67ab
evp 12268.28k 12572.67k 12626.77k 12727.98k 12926.98k 12899.67k


openssl speed -multi 1 -evp des-ede3-cfb1
Forked child 0
Got: +H:16:64:256:1024:8192:16384 from 0
Got: +F:22:des-ede3-cfb1:439258.67:440853.33:442026.67:428373.33:431300.66:431157.89 from 0
version: 3.0.0-alpha3-dev
built on: built on: Thu May 21 21:54:40 2020 UTC
options:bn(64,64) rc4(16x,int) des(int) aes(partial) idea(int) blowfish(ptr)
compiler: gcc -fPIC -pthread -m64 -Wa,--noexecstack -Wall -O3 -DOPENSSL_USE_NODELETE -DL_ENDIAN -DOPENSSL_PIC -DOPENSSL_BUILDI
NG_OPENSSL -DNDEBUG
CPUINFO: OPENSSL_ia32cap=0xfedaf387ffebffff:0x9c67ab
evp 439.26k 440.85k 442.03k 428.37k 431.30k 431.16k


openssl speed -multi 1 -evp des-ede-ecb
Forked child 0
Got: +H:16:64:256:1024:8192:16384 from 0
Got: +F:22:des-ede:29308330.67:30292181.33:30660096.00:31281152.00:30829226.67:29502122.67 from 0
version: 3.0.0-alpha3-dev
built on: built on: Thu May 21 21:54:40 2020 UTC
options:bn(64,64) rc4(16x,int) des(int) aes(partial) idea(int) blowfish(ptr)
compiler: gcc -fPIC -pthread -m64 -Wa,--noexecstack -Wall -O3 -DOPENSSL_USE_NODELETE -DL_ENDIAN -DOPENSSL_PIC -DOPENSSL_BUILDI
NG_OPENSSL -DNDEBUG
CPUINFO: OPENSSL_ia32cap=0xfedaf387ffebffff:0x9c67ab
evp 29308.33k 30292.18k 30660.10k 31281.15k 30829.23k 29502.12k
```

db71d315479762eefbf2bcda8be3b44b1867133f  - Performance dropped off 75%-90%+
```
openssl speed -multi 1 -evp aes256-wrap
Forked child 0
Got: +H:16:64:256:1024:8192:16384 from 0
Got: +F:22:id-aes256-wrap:-248.32:456.43:1080.33:454.26:311.89:158.88 from 0
version: 3.0.0-alpha3-dev
built on: built on: Thu May 21 22:09:45 2020 UTC
options:bn(64,64) rc4(16x,int) des(int) aes(partial) idea(int) blowfish(ptr)
compiler: gcc -fPIC -pthread -m64 -Wa,--noexecstack -Wall -O3 -DOPENSSL_USE_NODELETE -DL_ENDIAN -DOPENSSL_PIC -DOPENSSL_BUILDI
NG_OPENSSL -DNDEBUG
CPUINFO: OPENSSL_ia32cap=0xfedaf387ffebffff:0x9c67ab
evp -248.32 456.43 1080.33 454.26 311.89 158.88


openssl speed -multi 1 -evp id-aes256-wrap-pad
Forked child 0
Got: +H:16:64:256:1024:8192:16384 from 0
Got: +F:22:id-aes256-wrap-pad:-188.77:1264.68:265.41:804.73:210.51:191.09 from 0
version: 3.0.0-alpha3-dev
built on: built on: Thu May 21 22:09:45 2020 UTC
options:bn(64,64) rc4(16x,int) des(int) aes(partial) idea(int) blowfish(ptr)
compiler: gcc -fPIC -pthread -m64 -Wa,--noexecstack -Wall -O3 -DOPENSSL_USE_NODELETE -DL_ENDIAN -DOPENSSL_PIC -DOPENSSL_BUILDI
NG_OPENSSL -DNDEBUG
CPUINFO: OPENSSL_ia32cap=0xfedaf387ffebffff:0x9c67ab
evp -188.77 1264.68 265.41 804.73 210.51 191.09


openssl speed -multi 1 -evp id-aes128-wrap-pad
Forked child 0
Got: +H:16:64:256:1024:8192:16384 from 0
Got: +F:22:id-aes128-wrap-pad:-192.41:2713.78:3561.42:688.55:447.83:829.74 from 0
version: 3.0.0-alpha3-dev
built on: built on: Thu May 21 22:09:45 2020 UTC
options:bn(64,64) rc4(16x,int) des(int) aes(partial) idea(int) blowfish(ptr)
compiler: gcc -fPIC -pthread -m64 -Wa,--noexecstack -Wall -O3 -DOPENSSL_USE_NODELETE -DL_ENDIAN -DOPENSSL_PIC -DOPENSSL_BUILDI
NG_OPENSSL -DNDEBUG
CPUINFO: OPENSSL_ia32cap=0xfedaf387ffebffff:0x9c67ab
evp -192.41 2713.78 3561.42 688.55 447.83 829.74


openssl speed -multi 1 -evp des-ede3-cfb1
Forked child 0
Got: +H:16:64:256:1024:8192:16384 from 0
Got: +F:22:des-ede3-cfb1:-6.87:42.65:44.81:8.79:3.76:0.56 from 0
version: 3.0.0-alpha3-dev
built on: built on: Thu May 21 22:09:45 2020 UTC
options:bn(64,64) rc4(16x,int) des(int) aes(partial) idea(int) blowfish(ptr)
compiler: gcc -fPIC -pthread -m64 -Wa,--noexecstack -Wall -O3 -DOPENSSL_USE_NODELETE -DL_ENDIAN -DOPENSSL_PIC -DOPENSSL_BUILDI
NG_OPENSSL -DNDEBUG
CPUINFO: OPENSSL_ia32cap=0xfedaf387ffebffff:0x9c67ab
evp -6.87 42.65 44.81 8.79 3.76 0.56


openssl speed -multi 1 -evp des-ede-ecb
Forked child 0
Got: +H:16:64:256:1024:8192:16384 from 0
Got: +F:22:des-ede:-764.38:2039.30:2621.43:2935.81:2622.14:1183.84 from 0
version: 3.0.0-alpha3-dev
built on: built on: Thu May 21 22:09:45 2020 UTC
options:bn(64,64) rc4(16x,int) des(int) aes(partial) idea(int) blowfish(ptr)
compiler: gcc -fPIC -pthread -m64 -Wa,--noexecstack -Wall -O3 -DOPENSSL_USE_NODELETE -DL_ENDIAN -DOPENSSL_PIC -DOPENSSL_BUILDI
NG_OPENSSL -DNDEBUG
CPUINFO: OPENSSL_ia32cap=0xfedaf387ffebffff:0x9c67ab
evp -764.38 2039.30 2621.43 2935.81 2622.14 1183.84
```
e1c6f76281473b8fe66954187e793108a0e8568c  - Latest master commit. Performance is still bad.
```
openssl speed -multi 1 -evp aes256-wrap
Forked child 0
Got: +H:16:64:256:1024:8192:16384 from 0
Got: +F:22:id-aes256-wrap:-183.41:331.07:416.20:358.49:193.36:137.27 from 0
version: 3.0.0-alpha3-dev
built on: built on: Thu May 21 22:28:31 2020 UTC
options:bn(64,64) rc4(16x,int) des(int) aes(partial) idea(int) blowfish(ptr)
compiler: gcc -fPIC -pthread -m64 -Wa,--noexecstack -Wall -O3 -DOPENSSL_USE_NODELETE -DL_ENDIAN -DOPENSSL_PIC -DOPENSSL_BUILDI
NG_OPENSSL -DNDEBUG
CPUINFO: OPENSSL_ia32cap=0xfedaf387ffebffff:0x9c67ab
evp -183.41 331.07 416.20 358.49 193.36 137.27


openssl speed -multi 1 -evp id-aes256-wrap-pad
Forked child 0
Got: +H:16:64:256:1024:8192:16384 from 0
Got: +F:22:id-aes256-wrap-pad:-184.48:3909.15:592.60:396.11:360.46:142.25 from 0
version: 3.0.0-alpha3-dev
built on: built on: Thu May 21 22:28:31 2020 UTC
options:bn(64,64) rc4(16x,int) des(int) aes(partial) idea(int) blowfish(ptr)
compiler: gcc -fPIC -pthread -m64 -Wa,--noexecstack -Wall -O3 -DOPENSSL_USE_NODELETE -DL_ENDIAN -DOPENSSL_PIC -DOPENSSL_BUILDI
NG_OPENSSL -DNDEBUG
CPUINFO: OPENSSL_ia32cap=0xfedaf387ffebffff:0x9c67ab
evp -184.48 3909.15 592.60 396.11 360.46 142.25


openssl speed -multi 1 -evp id-aes128-wrap-pad
Forked child 0
Got: +H:16:64:256:1024:8192:16384 from 0
Got: +F:22:id-aes128-wrap-pad:-265.19:2063.38:675.46:2263.84:493.88:182.50 from 0
version: 3.0.0-alpha3-dev
built on: built on: Thu May 21 22:28:31 2020 UTC
options:bn(64,64) rc4(16x,int) des(int) aes(partial) idea(int) blowfish(ptr)
compiler: gcc -fPIC -pthread -m64 -Wa,--noexecstack -Wall -O3 -DOPENSSL_USE_NODELETE -DL_ENDIAN -DOPENSSL_PIC -DOPENSSL_BUILDI
NG_OPENSSL -DNDEBUG
CPUINFO: OPENSSL_ia32cap=0xfedaf387ffebffff:0x9c67ab
evp -265.19 2063.38 675.46 2263.84 493.88 182.50


openssl speed -multi 1 -evp des-ede3-cfb1
Forked child 0
Got: +H:16:64:256:1024:8192:16384 from 0
Got: +F:22:des-ede3-cfb1:-8.49:25.95:18.50:6.94:8.42:-0.01 from 0
version: 3.0.0-alpha3-dev
built on: built on: Thu May 21 22:28:31 2020 UTC
options:bn(64,64) rc4(16x,int) des(int) aes(partial) idea(int) blowfish(ptr)
compiler: gcc -fPIC -pthread -m64 -Wa,--noexecstack -Wall -O3 -DOPENSSL_USE_NODELETE -DL_ENDIAN -DOPENSSL_PIC -DOPENSSL_BUILDI
NG_OPENSSL -DNDEBUG
CPUINFO: OPENSSL_ia32cap=0xfedaf387ffebffff:0x9c67ab
evp -8.49 25.95 18.50 6.94 8.42 -0.01


openssl speed -multi 1 -evp des-ede-ecb
Forked child 0
Got: +H:16:64:256:1024:8192:16384 from 0
Got: +F:22:des-ede:-542.86:3544.15:957.14:968.85:3503.97:1378.97 from 0
version: 3.0.0-alpha3-dev
built on: built on: Thu May 21 22:28:31 2020 UTC
options:bn(64,64) rc4(16x,int) des(int) aes(partial) idea(int) blowfish(ptr)
compiler: gcc -fPIC -pthread -m64 -Wa,--noexecstack -Wall -O3 -DOPENSSL_USE_NODELETE -DL_ENDIAN -DOPENSSL_PIC -DOPENSSL_BUILDI
NG_OPENSSL -DNDEBUG
CPUINFO: OPENSSL_ia32cap=0xfedaf387ffebffff:0x9c67ab
evp -542.86 3544.15 957.14 968.85 3503.97 1378.97
```

### PR DESCRIPTION
**Pull request by t8m**
The __TMS might be necessary on VMS however there is no such
define on glibc even though the times() function is fully supported.
    
Fixes #11903

### CODE REVIEW

File: apps/lib/apps.c [hunk 1]
```diff
@@ -2252,7 +2252,7 @@ double app_tminterval(int stop, int usertime)
         struct timeval tv;
         gettimeofday(&tv, NULL);
         now = (clock_t)((unsigned long long)tv.tv_sec * CLK_TCK +
-                        (unsigned long long)tv.tv_usec * (1000000 / CLK_TCK)
+                        (unsigned long long)tv.tv_usec * (CLK_TCK / 1000000)
```
Comment (levitte) [inline apps/lib/apps.c:2255]: Oh!!!!!

Comment (kroeckx) [reply apps/lib/apps.c:2255]: I've noticed that too, but this is still wrong. You will now get 0, you need to remove the ().

**Comment by levitte** [comment]
This should be back-ported to 1.1.1 as well.

File: apps/lib/apps.c (hunk 1 above)
Comment (kroeckx) [reply apps/lib/apps.c:2255]: Please also fix the uses of () for the if above.

**Comment by levitte** [comment]
I just read up on the VMS CC release notes, and as far as I can tell, all the POSIX-1 functions are supported since at least VMS 7.1, which is our minimum requirement since OpenSSL 1.1.0, so it means that you can simply remove the whole related VMS-specific section.

**Comment by kroeckx** [comment]
Is that including the sysconf(_SC_CLK_TCK)?

**Comment by levitte** [comment]
> Is that including the sysconf(_SC_CLK_TCK)?

https://vmssoftware.com/pdfs/HP_branded_docs_1st_batch/HP_C_RTL_reference_manual.pdf says (in Appendix A) that `sysconf` is available since VMS 7.0, and doesn't say anything special about `_SC_CLK_TCK`, so I believe it's safe to assume that is was there from then on.

File: apps/lib/apps.c (hunk 1 above)
Comment (levitte) [reply apps/lib/apps.c:2255]: Thanks @kroeckx.  I just had another look at the definition of `CLK_TCK` on VMS.  It's 100, which had me wonder "what the hell?????"

Comment (kroeckx) [reply apps/lib/apps.c:2255]: Note that it's only used for -elapsed, so you probably won't notice it's weird behaviour with things like negative time.

**Comment by t8m** [comment]
OK, I am going to remove the VMS #ifdef.

**Comment by t8m** [comment]
Please re-approve.

**Comment by t8m** [comment]
Merged to master.

**Pull request closed without merging**